aws-auth-cognito/src/main/java/com/amplifyframework/statemachine/codegen/data/AuthConfiguration.kt

```diff
@@ -15,28 +15,70 @@
 
 package com.amplifyframework.statemachine.codegen.data
 
+import androidx.annotation.IntRange
+import com.amplifyframework.annotations.InternalAmplifyApi
+import com.amplifyframework.auth.AuthUserAttributeKey
 import com.amplifyframework.auth.cognito.options.AuthFlowType
+import org.json.JSONArray
 import org.json.JSONObject
 
+private const val configName = "Default"
+
+@InternalAmplifyApi
+enum class VerificationMechanism {
+    PhoneNumber, Email
+}
+
+@InternalAmplifyApi
+data class PasswordProtectionSettings(
+    @IntRange(from = 6, to = 99) val length: Int,
+    val requiresNumber: Boolean,
+    val requiresSpecial: Boolean,
+    val requiresUpper: Boolean,
+    val requiresLower: Boolean
+)
```

Comment:
nice, i like that this data type is closer to the new config schema's type. I had opted to make less changes in the Authenticator by just moving over the Authenticator's PasswordProtectionSettings type over to Amplify, so the difference is this https://github.com/aws-amplify/amplify-swift/pull/3566/files#diff-8cfb4aa001ce3e3a0d142cf84c5371705975660e54b372128d159115ea425220

Reply:
I also moved this directly (just renamed the class) - on Android we were already using a class of this structure 🤷